app/models/color_scheme.rb

```diff
@@ -318,6 +318,14 @@ def self.hex_cache
   scope :without_theme_owned_palettes,
         -> { where("color_schemes.id NOT IN (SELECT color_scheme_id FROM theme_color_schemes)") }
 
+  scope :with_experimental_system_theme_palettes,
+        -> do
+          joins("LEFT JOIN themes on themes.id = color_schemes.theme_id").where(
+            "themes.id is NULL OR themes.id > 0 OR LOWER(themes.name) IN (?)",
+            SiteSetting.experimental_system_themes_map,
```

Comment:
Just a thought, rather than relying on names, you could add a method to `Theme` like `Theme.experimental_system_theme_ids`

```ruby
def experimental_system_theme_ids
  Theme::CORE_THEMES.select { |k, v| SiteSetting.experimental_system_themes_map.include?(k) }.values
end
```

Then you could do `themes.id is NULL OR themes.id > 0 OR themes.id IN (?)`